File: frontend/lib/dto/card_dto/TravelPlanResponse.dart
class TravelPlanResponse {
  final DateTime startDate;
  final DateTime endDate;
  final List<String> countries;
  final String status;
  final double travelLimit;
  final int maxDays;

  TravelPlanResponse({
    required this.startDate,
    required this.endDate,
    required this.countries,
    required this.status,
    required this.travelLimit,
    required this.maxDays,
  });

  factory TravelPlanResponse.fromJson(Map<String, dynamic> json) {
    return TravelPlanResponse(
      startDate: DateTime.parse(json['startDate']),
      endDate: DateTime.parse(json['endDate']),
      countries: List<String>.from(json['countries']),
      status: json['status'],
      travelLimit: json['travelLimit']?.toDouble() ?? 0.0,
      maxDays: json['maxDays'] ?? 0,
    );
  }
}

File: microservices/travel-plan-service/src/main/java/hps/ma/travelplanservice/dto/TravelPlanResponse.java
package hps.ma.travelplanservice.dto;

import hps.ma.travelplanservice.dao.enums.TravelPlanStatus;
import lombok.*;

import java.util.Date;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class TravelPlanResponse {
    private Long id;
    private Date startDate;
    private Date endDate;
    private List<String> countries;
    private TravelPlanStatus status;
    private Double travelLimit;
    private int maxDays;
    private Long cardId;
}